### PR TITLE
consolidate numeric vectors

### LINKED ISSUE #7639: EnumerateLibrary.GetPosition() surprisingly slow
**Describe the bug**
In a simple enumeration loop, if one is iterating over library.GetPosition() in the loop, it takes up to 40% of the runtime of the loop.
For a full enumeration, counting/calculating the position yourself everytime in python is much faster and reduces the overhead to almost zero compared to the actual work in the iterator (i.e. RunReactants).

**To Reproduce**
Will update to a full example soon.

```
    library = EnumerateLibrary(rxn, bbs)
    reagents = library.GetReagents()
    for product_template_results in library:
        # print input reactant monomers and the product
        libpos = library.GetPosition()
        for i, n in enumerate(libpos): # <- this line iterates over a vector of size_t with 3 elements and takes 40% of the runtime
            print(reagents[i][n].GetProp("_Name"))
```

**Expected behavior**
The line taking less than 1% of the runtime.

**Screenshots**
Screenshot 2024-07-20 at 11 26 53

**Configuration (please complete the following information):**
 - RDKit version: 2023.09.6
 - OS: macOS 14.5
 - Python version (if relevant): 3.11
 - Are you using conda? no

### PR DESCRIPTION
**Pull request by bp-kelley**
This consolidates the numeric wrappers to RDBoost so that whey get the `__iter__` fixes.

This resolves #7639

### CODE REVIEW

File: rdkit/Chem/FilterCatalog.py
Comment (greglandrum) [inline rdkit/Chem/FilterCatalog.py:1]: Why are the changes in this file necessary?

Comment (bp-kelley) [reply rdkit/Chem/FilterCatalog.py:1]: An overabundance of caution.  MatchVectType used to be rdkit.Chem.FilterCatalog.MatchVectType and is used in one of the tests.

Comment (bp-kelley) [reply rdkit/Chem/FilterCatalog.py:1]: @greglandrum did this answer make sense?

**Comment by greglandrum** [comment]
In my hands applying this PR has no effect on the runtime of the two sets of tests we've been using so far:

```
from rdkit import Chem
m = Chem.MolFromSmiles('C'*50)
ps = Chem.FindAllPathsOfLengthN(m,3)
%timeit [(i,x) for i,x in enumerate(ps)]
ps47 = Chem.FindAllPathsOfLengthN(m,47)
%timeit [(i,x) for i,x in enumerate(ps47)]
```
and
```
from rdkit import Chem
from rdkit.Chem import AllChem
from rdkit.Chem import rdChemReactions

for N in range(3, 8):
    carbons = [f"[C:{i}]" for i in range(1,N+1)]
    rxn = f"{'.'.join(carbons)}>>{''.join(carbons)}"
    rxn = AllChem.ReactionFromSmarts(rxn)
    mol = Chem.MolFromSmiles("C")
    bbs = []
    for i in range(N):
        bbs.append( [mol] * 5 )
        
    lib = rdChemReactions.EnumerateLibrary(rxn, bbs)
    print(f"{N=}", len(lib.GetPosition()), end=" ")
    %timeit [x for x in lib.GetPosition()]
```

where did you see a performance gain with these changes?

File: rdkit/__init__.py [hunk 1]
```diff
@@ -59,8 +59,10 @@ def __next__(self):
 def __vect__iter__(vect):
     return VectIter(vect)
 
+VECT_WRAPS = {'MatchTypeVect', 'UnsignedLong_Vect', 'VectSizeT', 'VectorOfStringVectors'}
```
Comment (bp-kelley) [inline rdkit/__init__.py:62]: @greglandrum these have the default SWIG names, so I had to hard code them.

Should be good to go now,

**Review by greglandrum — APPROVED** [review]
LGTM

**Pull request merged**